Core/Src/main.c

```diff
@@ -0,0 +1,956 @@
+  MX_GPIO_Init();
+  MX_DMA_Init();
+  MX_TIM1_Init();
+  MX_USB_Device_Init();
+  MX_TIM2_Init();
+  MX_TIM16_Init();
+  MX_TIM3_Init();
+  MX_TIM4_Init();
+  MX_TIM8_Init();
+  MX_TIM6_Init();
+  MX_TIM15_Init();
+  /* USER CODE BEGIN 2 */
+    /*uint32_t esc_dshot_buffer[5] = {25, 50, 0, 25, 0};
+  */
+    //Motor motor = {&htim2, TIM_CHANNEL_1};
+    //dshot_init(DSHOT150, &motor);
+    //  htim2.Instance->CCR1 = 50;
+    //  HAL_TIM_PWM_Start(&htim2, TIM_CHANNEL_1);
+    //
+    //   HAL_Delay(5000);
+  /* USER CODE END 2 */
+
+  /* Infinite loop */
+  /* USER CODE BEGIN WHILE */
+
+    DWT_Delay_Init();
+
+
+    dshot600(motor, 0);
+    HAL_TIM_PWM_Start_DMA(&htim15, TIM_CHANNEL_1, motor, DSHOT_FRAME_SIZE);
+    HAL_Delay(2000);
+    dshot600(motor, 100);
+    HAL_Delay(1000);
+    //        for(int i = 1024; i<2048; i++) {
+    //            dshot600(motor, i);
+    //            HAL_TIM_PWM_Start_DMA(&htim8, TIM_CHANNEL_1, motor, DSHOT_FRAME_SIZE);
+    //            HAL_Delay(50);
+    //        }
+
+    HAL_TIM_Encoder_Start(&htim1, TIM_CHANNEL_ALL);
+    HAL_TIM_Encoder_Start(&htim2, TIM_CHANNEL_ALL);
+    HAL_TIM_Encoder_Start(&htim3, TIM_CHANNEL_ALL);
+    //HAL_TIM_Base_Start_IT(&htim6);
+
+    HAL_GPIO_WritePin(nSleep_GPIO_Port, nSleep_Pin, 1);
+    HAL_GPIO_WritePin(M1_DIR_GPIO_Port, M1_DIR_Pin, 1);
+    //HAL_TIM_PWM_Start(&htim4, TIM_CHANNEL_1);
+    HAL_TIM_PWM_Start(&htim4, TIM_CHANNEL_3);
+    HAL_TIM_PWM_Start(&htim4, TIM_CHANNEL_1);
+    HAL_TIM_PWM_Start(&htim16, TIM_CHANNEL_1);
+    TIM4->CCR3  = 0;
+    TIM4->CCR1  = 0;
+    TIM16->CCR1 = 0;
+
+    HAL_TIM_Base_Start_IT(&htim6);
+
+#pragma clang diagnostic push
+#pragma ide diagnostic ignored "EndlessLoop"
+    while (1) {
+
+        //call();
+
+
+        //dshot600(motor, ((TIM3->CNT)>>2)/2048);
+        //        *motor = prepareDshotPacket(((TIM3->CNT)>>2)/2048);
+        //        HAL_TIM_PWM_Start_DMA(&htim2, TIM_CHANNEL_1, motor, DSHOT_FRAME_SIZE);
+
+
+        struct Data data = {1, motorSpeeds[0], motorSpeeds[1], motorSpeeds[2]};
+        //struct Data data1 = {2, DT, DT >> 8, 8};
+
+        //USBD_CUSTOM_HID_SendReport(&hUsbDeviceFS, (uint8_t *) &data, 4);
+        if (DT != 0) {
+            //USBD_CUSTOM_HID_SendReport(&hUsbDeviceFS, (uint8_t *) &data1, sizeof(struct Data));
+            dshot600(motor, DT);
+            HAL_TIM_PWM_Start_DMA(&htim15, TIM_CHANNEL_1, motor, DSHOT_FRAME_SIZE);
+            DT = 0;
+            HAL_Delay(100);
+            struct Data data2 = {2, DT, DT >> 8, 8};
+            //USBD_CUSTOM_HID_SendReport(&hUsbDeviceFS, (uint8_t *) &data2, sizeof(struct Data));
+        }
+        if(motor_new){
+            //USBD_CUSTOM_HID_SendReport(&hUsbDeviceFS, (uint8_t *) &data, sizeof(struct Data));
+            motor_new = 0;
+            HAL_GPIO_WritePin(nSleep_GPIO_Port, nSleep_Pin, 0);
+            DWT_Delay_us(30);
+            HAL_GPIO_WritePin(nSleep_GPIO_Port, nSleep_Pin, 1);
+            HAL_Delay(100);

```

Comment:
There is no need for delay after setting nSleep high. 100 ms delay also blocks the while loop for too long.